osquery/tables/events/windows/windows_events.cpp

```diff
@@ -6,138 +6,246 @@
  *  the LICENSE file found in the root directory of this source tree.
  */
 
-#include <string>
+#include <sstream>
 
 #include <boost/algorithm/string.hpp>
 #include <boost/property_tree/json_parser.hpp>
-#include <boost/property_tree/ptree.hpp>
-#include <boost/property_tree/xml_parser.hpp>
 
-#include <osquery/config/config.h>
-#include <osquery/core.h>
 #include <osquery/flags.h>
 #include <osquery/logger.h>
 #include <osquery/registry_factory.h>
-#include <osquery/tables.h>
-
-#include <osquery/events/windows/windows_event_log.h>
-#include <osquery/filesystem/fileops.h>
+#include <osquery/tables/events/windows/windows_events.h>
 #include <osquery/utils/conversions/split.h>
-#include <osquery/utils/conversions/windows/strings.h>
-#include <osquery/utils/json/json.h>
-
-namespace pt = boost::property_tree;
+#include <osquery/utils/conversions/tryto.h>
 
 namespace osquery {
+namespace {
+std::string normalizeChannelName(std::string channel) {
+  boost::erase_all(channel, "\"");
+  boost::erase_all(channel, "\'");
+
+  std::transform(channel.begin(), channel.end(), channel.begin(), ::tolower);
+  return channel;
+}
+} // namespace
+
+FLAG(bool, enable_wel_events, false, "Enables Windows Event Log events");
```

Comment:
Oofda. That seems a bit confusing to me. `enable_wel_events`(subscriber) _and_ `enable_windows_events`(publisher)? What's the difference? _Both_ need to be on for us to get a steady stream of WEL right? I get the design decision as now one could have just powershell events with the WEL publisher, but maybe there's a better naming convention we could adopt? Perhaps `enable_windows_events_publisher` and `enable_windows_events_subscriber`? Then we'd also change the powershell subscriber to `enable_powershell_events_subscriber`? I definitely prefer being terse if possible, but in this case it might be beneficial to be a bit more verbose. What do you think?

Reply:
The flags are being introduced in an effort to standardize how publishers/subscribers are enabled, so that performance-wise administrators only pay for what they explicitly enable. We have recently looked at a large Windows deployment where the fleet management software in use was (in its default configuration) passing the `--disable_events=false` flag to all the enrolled hosts. Windows devices would automatically enable the publishers and subscribers even though there were no queries scheduled against the tables, causing mass event expiration which would add additional heavy load on the database.

The names you have proposed look great, and I'll update the code to use them!